packages/cli/src/controllers/api-keys.controller.ts

```diff
@@ -19,18 +23,36 @@ export class ApiKeysController {
 	constructor(
 		private readonly eventService: EventService,
 		private readonly publicApiKeyService: PublicApiKeyService,
+		private readonly apiKeysRepository: ApiKeyRepository,
+		private readonly license: License,
 	) {}
 
 	/**
 	 * Create an API Key
 	 */
 	@Post('/', { middlewares: [isApiEnabledMiddleware] })
-	async createAPIKey(req: AuthenticatedRequest) {
-		const newApiKey = await this.publicApiKeyService.createPublicApiKeyForUser(req.user);
+	async createAPIKey(
+		req: AuthenticatedRequest,
+		_res: Response,
+		@Body payload: CreateOrUpdateApiKeyRequestDto,
+	) {
+		const currentNumberOfApiKeys = await this.apiKeysRepository.countBy({ userId: req.user.id });
+
+		if (currentNumberOfApiKeys >= this.license.getApiKeysLimit()) {
+			throw new BadRequestError('You have reached the maximum number of API keys allowed.');
+		}
+
+		const newApiKey = await this.publicApiKeyService.createPublicApiKeyForUser(req.user, {
+			withLabel: payload.label,
```

Comment:
naming here could be clearer. `withLabel` sounds like it should be a boolean in my head. Here we are passing the label itself
```suggestion
			label: payload.label,
```